app/models/score.rb

```diff
@@ -1,10 +1,20 @@
 class Score < ApplicationRecord
 
-  def self.high_score(user)
-    where(user_id: user).order(score: :desc).first
-  end
-  
-  def self.top_scores_by_location(geo_scope, user_location)
-    Score.where("#{geo_scope} = '#{user_location}'").order(score: :desc).limit(5)
+  # def self.high_score(user)
+  #   where(user_id: user).order(score: :desc).first
+  # end
+  #
+  # def self.top_scores_by_location(geo_scope, user_location)
+  #   where("#{geo_scope} = '#{user_location}'").order(score: :desc).limit(5)
+  # end
+
+  def self.top_5_highest_scores(params)
+    if params.has_key?(:geo_scope)
+      where("#{params[:geo_scope]} = '#{params[:user_location]}'").order(score: :desc).limit(5)
+    elsif params.has_key?(:user_id)
+      where(user_id: params[:user_id]).order(score: :desc).limit(5)
+    else
+      order(score: :desc).limit(5)
+    end
```

Comment:
this method looks good for now, but may need some extra conditionals to account for a sad path where someone sends both params, for example
(of course, we're the only ones consuming the API, so maybe it's not necessary to play for that? not sure what the standard should be here)